admin/services/marking.service.js

```diff
@@ -43,23 +44,35 @@ markingService.mark = async function (completedCheck) {
     processedAt: moment.utc()
   }
 
+  const checkForm = await checkFormDataService.sqlFindOneById(completedCheck.checkForm_id)
```

Comment:
The checkForm should not be retrieved from inside the loop - please fetch all checkForms for the batch in one query before the loop, and pass the relevant checkForm in.

Reply:
@GeorgeChatzigiannis this has come up a few times.  You really need to think carefully about the performance implications

Reply:
@jon-shipley @GuyHarwood that was a naive mistake from my end, it won't happen again.
This has been updated.